src/streaming/controllers/PlaybackController.js

```diff
@@ -257,7 +257,8 @@ function PlaybackController() {
      */
     function getStreamStartTime(ignoreStartOffset) {
         let presentationStartTime;
-        let startTimeOffset = !ignoreStartOffset ? parseInt(URIQueryAndFragmentModel(context).getInstance().getURIFragmentData().s, 10) : NaN;
+        let fragData = URIQueryAndFragmentModel(context).getInstance().getURIFragmentData();
+        let startTimeOffset = !ignoreStartOffset ? parseInt(fragData.s || fragData.t, 10) : NaN;
```

Comment:
>> -        let startTimeOffset = !ignoreStartOffset ? parseInt(URIQueryAndFragmentModel(context).getInstance().getURIFragmentData().s, 10) : NaN;
>> +        let fragData = URIQueryAndFragmentModel(context).getInstance().getURIFragmentData();
>> +        let startTimeOffset = !ignoreStartOffset ? parseInt(fragData.s || fragData.t, 10) : NaN;

>if fragData.s value is set to 0 it will not set and use the other side of
the || and look for fragData.t ....I believe.

>Keep seeing this happen all over the code with things that could be set to
0 and we use something || >something else.

Agree it's not a good pattern and we (Lloyd) should fix it, but fragData is
always a string in the current implementation so "" would be false which is
fine, but "0" would be true, which would be handled correctly.

On Fri, May 13, 2016 at 3:55 AM, Dan Sparacio <notifications@github.com>
wrote:

> In src/streaming/controllers/PlaybackController.js
> <https://github.com/Dash-Industry-Forum/dash.js/pull/1390#discussion_r63127507>
> :
>
> > @@ -257,7 +257,8 @@ function PlaybackController() {
> >       */
> >      function getStreamStartTime(ignoreStartOffset) {
> >          let presentationStartTime;
> > -        let startTimeOffset = !ignoreStartOffset ? parseInt(URIQueryAndFragmentModel(context).getInstance().getURIFragmentData().s, 10) : NaN;
> > +        let fragData = URIQueryAndFragmentModel(context).getInstance().getURIFragmentData();
> > +        let startTimeOffset = !ignoreStartOffset ? parseInt(fragData.s || fragData.t, 10) : NaN;
>
> if fragData.s value is set to 0 it will not set and use the other side of
> the || and look for fragData.t ....I believe.
>
> Keep seeing this happen all over the code with things that could be set to
> 0 and we use something || something else.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Dash-Industry-Forum/dash.js/pull/1390/files/ae0f2ef978b79a1cb51a88b8780c38104a8315aa#r63127507>
>